test/test_opaque_obj_v2.py

```diff
@@ -553,6 +619,157 @@ def test_invalid_schema(self):
                 lib=self.lib,
             )
 
+    def test_invalid_opaque_obj_types(self):
+        for t in [str, bool, int, float, torch.Tensor]:
+            with self.assertRaisesRegex(ValueError, "Unable to register built-in type"):
+                register_opaque_type(t)
+
+        @dataclass
+        class Bad1:
+            x: int
+
+        pytree.register_dataclass(Bad1)
+        with self.assertRaisesRegex(
+            ValueError, "cannot be registered as an opaque object"
+        ):
+            register_opaque_type(Bad1)
+
+        @dataclass
+        class Bad2:
+            x: int
+
+        register_opaque_type(Bad2)
+        with self.assertRaisesRegex(ValueError, "cannot be registered as a pytree"):
+            pytree.register_dataclass(Bad2)
```

Comment:
what's the rule here? something can only be registered as either a pytree or opaque object but not both?

If so, can include more of the error message or leave a comment

Reply:
yes, we cant register it as a pytree if it's already an opaque object and vise versa. updated the error msg!